plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -360,6 +361,181 @@ struct block_time_tracker {
    bool paused = false;
 };
 
+/**
+ * Tracks active finalizers and votes via production_pause_vote_tracker to determine if implicit pause
+ * of production is needed.
+ */
+struct implicit_production_pause_vote_tracker {
+
+   void init(const controller& chain,
+             const std::set<chain::account_name>& producers,
+             fc::microseconds production_pause_vote_timeout)
+   {
+      _chain = &chain;
+      _producers = producers; // copy, normally contains at most one
+      _vt.set_vote_timeout(production_pause_vote_timeout);
+   }
+
+   // Only call when savanna is active
+   void activate() {
+      // disabled by configuration
+      if (_production_pause_vote_timeout.count() == 0)
+         return;
+      // disable implicit pause for eosio
+      if (_producers.contains(eosio::chain::config::system_account_name))
+         return;
+      // no producers configured so there is no possible production to pause
+      if (_producers.empty())
+         return;
+
+      _active = true;
+   }
```

Comment:
I don't think the `activate` function is necessary and it would be better to remove it.

But if we did want to keep it for some reason, the implementation should be:

```cpp
void activate() {
   const auto vtm = get_vote_track_mode();

   if (vtm == vote_track_mode::inactive) {
      // Change to `vote_track_mode::all_votes` for now so that a future call to
      // `update_active_finalizers` has an opportunity to set it correctly going forward.
      _vote_track_mode.store(vote_track_mode::all_votes, std::memory_order_relaxed);
   }
   // Otherwise:
   // It is disabled, in which case we do not want to activate it.
   // Or it is already activated, in which case we do not want to change the vote_track_mode.
}
```